### PR TITLE
Allow adding plugin to express-pouchdb

### PR DESCRIPTION
**Pull request by siuying**
I'm trying to write an extension to allow using JSON Web Token to authenticate access.

The plugin itself is very simple and I can get it work in no time (https://github.com/siuying/express-pouchdb/blob/jwt/lib/routes/jwt.js). But when I try to extract it into separate plugin, they didnt work as expected. 

I think the problem is the plugin needs `config-infrastructure`, and also need to load before remaining routes. I figure having ability to insert plugin into the `express-pouchdb` might be helpful for others as well, would you please see if it is suitable to merge?

### CODE REVIEW

**Comment by marten-de-vries** [comment]
Your implementation seems a relatively elegant extension of the mode system. I wouldn't be opposed to merging this, but I'd like someone else to weigh in on that too (@nolanlawson , @daleharvey ?, someone else?).

It would be nice to have it documented in README.md, and have a test added to https://github.com/pouchdb/express-pouchdb/blob/master/test/test.js , but it wouldn't be a deal breaker for me. In any case, hold off on that until someone else gives the go ahead for the future.

File: lib/index.js [hunk 1]
```diff
@@ -185,14 +187,25 @@ module.exports = function (startPouchDB, opts) {
...
-    'replicator',
+    'replicator'
+  ].forEach(function (file) {
+    if (app.includes[file]) {
+      require('./' + file)(app);
+    }
+  });
+
+  app.plugins.forEach(function (plugin) {
+    require(plugin)(app)
```
Comment (marten-de-vries) [inline lib/index.js]: missing semicolon, this makes Travis fail.

**Comment by marten-de-vries** [comment]
Also: if this gets merged and you publish your plugin, it would be nice to have it mentioned in the readme so others can make use of it too - just like the collection of PouchDB plugins.

**Comment by nolanlawson** [comment]
Yes, at a minimum I would prefer to see some README documentation proposing the change. As for the change itself, it seems to add minimal risk to the codebase (and I'm +1 on the specific implementation), but again, it needs generous documentation please. :)

**Comment by nolanlawson** [comment]
Tests are borked due to a missing semicolon BTW.

**Comment by siuying** [comment]
I fixed the semicolon, and added some docs that should get people started. The test still fail, but it seems cause by something else

**Comment by nolanlawson** [comment]
Yeah, `pouchdb` as a dependency is borked since 5.2.0. I need to investigate.

**Comment by marten-de-vries** [comment]
@siuying: @nolanlawson fixed the tests, if you rebase Travis should be able to run properly. That should make us able to move forward on this.

**Comment by siuying** [comment]
done!

**Comment by garrensmith** [comment]
@siuying could you rebase this and then see if the tests all pass.

**Comment by nolanlawson** [comment]
Sorry, closing old PRs that haven't seen much activity in awhile. Seems this would need a bit more work in order to get merged, please reopen as necessary.

**Pull request closed without merging**